presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxMetadata.java

```diff
@@ -114,33 +121,61 @@ private JmxTableHandle getJmxHistoryTableHandle(SchemaTableName tableName)
+            String objectNamePattern = toPattern(tableName.getTableName()); //toLowerCase(ENGLISH));
+            List<ObjectName> objectNames = mbeanServer.queryNames(WILDCARD, null).stream()
+                    .filter(name -> name.getCanonicalName().toLowerCase(ENGLISH).matches(objectNamePattern))
+                    .collect(toImmutableList());
+            if (objectNames.isEmpty()) {
                 return null;
             }
-            MBeanInfo mbeanInfo = mbeanServer.getMBeanInfo(objectName.get());
-
-            ImmutableList.Builder<JmxColumnHandle> columns = ImmutableList.builder();
+            List<JmxColumnHandle> columns = new ArrayList<>();
             columns.add(new JmxColumnHandle(NODE_COLUMN_NAME, createUnboundedVarcharType()));
+            columns.add(new JmxColumnHandle(OBJECT_NAME_NAME, createUnboundedVarcharType()));
+            for (ObjectName objectName : objectNames) {
+                MBeanInfo mbeanInfo = mbeanServer.getMBeanInfo(objectName);
+
+                getColumnHandles(mbeanInfo).forEach(columns::add);
+            }
 
             // Since this method is being called on all nodes in the cluster, we must ensure (by sorting)
             // that attributes are in the same order on all of them.
-            Arrays.stream(mbeanInfo.getAttributes())
-                    .filter(MBeanAttributeInfo::isReadable)
-                    .map(attribute -> new JmxColumnHandle(attribute.getName(), getColumnType(attribute)))
+            columns = columns.stream()
+                    .distinct()
                     .sorted(comparing(JmxColumnHandle::getColumnName))
-                    .forEach(columns::add);
+                    .collect(toImmutableList());
 
-            return new JmxTableHandle(tableName, ImmutableList.of(objectName.get().toString()), columns.build(), true);
+            return new JmxTableHandle(tableName, objectNames.stream().map(ObjectName::toString).collect(toImmutableList()), columns, true);
         }
         catch (JMException e) {
             return null;
         }
     }
 
+    private String toPattern(String tableName)
+            throws MalformedObjectNameException
+    {
+        if (!tableName.contains("*")) {
+            return new ObjectName(tableName).getCanonicalName();
+        }
+        String pattern = Pattern.compile("\\*").splitAsStream(tableName)
+                .map(Pattern::quote)
+                .collect(Collectors.joining(".*"));
+        if (tableName.startsWith("*")) {
+            pattern = ".*" + pattern;
+        }
+        if (tableName.endsWith("*")) {
+            pattern += ".*";
+        }
+        return pattern;
+    }
+
+    private Stream<JmxColumnHandle> getColumnHandles(MBeanInfo mbeanInfo)
+    {
+        return Arrays.stream(mbeanInfo.getAttributes())
+                .filter(MBeanAttributeInfo::isReadable)
+                .map(attribute -> new JmxColumnHandle(attribute.getName(), getColumnType(attribute)));
```

Comment:
Will we not end up with multiple columns with same name here?

I guess we should remove duplicates from target list.

Based on example in the doc, you are doing it. Where is that happening?

Reply:
They are removed here: 

```
   columns = columns.stream()
                    .distinct()
                    .sorted(comparing(JmxColumnHandle::getColumnName))
                    .collect(toImmutableList());
```